day1/lib/day1.rb

```diff
@@ -0,0 +1,62 @@
+class Day1
+  attr_accessor :expense_report
+
+  def initialize(expense_report)
+    @expense_report = expense_report
+  end
+
+  def self.from_file(filepath)
+    expense_report_data = []
+    File.foreach(filepath).each do |line|
+      expense_report_data.append(line.chomp.to_i)
+    end
+    new(expense_report_data)
+  end
```

Comment:
This can be condensed to a single flow with no locals:

```ruby
  def self.from_file(filepath)
    new(File.foreach(filepath).map { |line| line.chomp.to_i })
  end

Reply:
Good stuff, thank you. Still unpacking the power of `map`.